### ISSUE TITLE
OutputWriter does not respect Symfony/Console OutputInterface

### ISSUE THREAD

**Issue by greeny** [Zenify/DoctrineMigrations#25]
If I set NullOutput to Symfony/Console, migrations still output text, because of OutputWriter not respecting it.

**Comment by TomasVotruba** [comment]
Could you send PR?

**Comment by greeny** [comment]
This is little bit tricky.

The thing is you need to register `Configuration` in DI container and therefore pass `$outputWriter` there. If you pass `NULL`, then default `OutputWriter` is created (which basically does nothing). However, it still /is created, so when command gets executed and asks for OutputWriter (https://github.com/doctrine/migrations/blob/master/lib/Doctrine/DBAL/Migrations/Tools/Console/Command/AbstractCommand.php#L139), it returns that default OutputWriter (which is definitely the case why you created your own OutputWriter). But we need this method to actually create OutputWriter for us (it has factory using `OutputInterface`).

So basically there is no soultion I can think off.

I first thought we could extend all the command classes and override some of the methods, that work with OutputWriter, but without success (`getOutputWriter` and `$outputWriter` are both private, and there is no setter). Of course we can **hack** it that it would use another property, e.g. `$diOutputWriter` instead of `$outputWriter`, but I don't like much this soultion.

What do you think about this? Or is there a solution I missed?

**Comment by TomasVotruba** [comment]
Could you also paste mininal code, that fails?

**Comment by greeny** [comment]
```php
$migrateCommand = new ArrayInput([
	'command' => 'migrations:migrate',
	'--no-interaction',
	'--allow-no-migration',
]);
$application = $container->getByType(Kdyby\Console\Application::class);
$application->run($migrateCommand, NULL /*or new NullOutput*/);
```

**Comment by TomasVotruba** [comment]
Thanks!

**Comment by TomasVotruba** [comment]
Just testing your code.

If you use `NULL` as second argument, `ConsoleOutput` [will be created by default](https://github.com/symfony/symfony/blob/fe6e9e4543d2a35c4adf7d423ef82680f3e6d445/src/Symfony/Component/Console/Application.php#L117).
 
```php
use Symfony\Component\Console\Application;

$migrateCommand = new ArrayInput(['command' => 'migrations:status']);

$container = (new ContainerFactory)->create();

/** @var Application $application */
$application = $container->getByType(Application::class);

$nullOutput = new NullOutput;
$application->run($migrateCommand, $nullOutput);
```

And the output is gone. If I use `NULL`, the code is shown.

Could you reproduce?

**Comment by TomasVotruba** [comment]
I've tried exactly your command:

```
$migrateCommand = new ArrayInput([
	'command' => 'migrations:migrate',
	'--no-interaction',
	'--allow-no-migration',
]);
```

...and there IS some output. It looks like it uses two different approaches even inside migrations.

**Comment by greeny** [comment]
Lets take an example: command `migrations:migrate`.

There is two way of outputting stuff in this command:

- Using `OutputInterface` (the banner "Doctrine Migrations")
- Using `OutputWriter` (SQL commands)

The first thing works correctly (when passed `NullOutput`, there is no banner), but SQL commands are written every time, not dependent on `OutputInterface`.

------

For different commands I did not check what is written by which method, but I quess it is not used only in migrate command.

**Comment by TomasVotruba** [comment]
I've tried to send fix: https://github.com/Zenify/DoctrineMigrations/commit/f273e85d7396bcdfbf3c7ac75a1faf15ed69ee97

Could you try it?

**Comment by greeny** [comment]
Seems like its working fine. Thanks for it :) I will let you know if I run into any problems with this.

**Comment by TomasVotruba** [comment]
Great! Happy to help :)

**Comment by TomasVotruba** [comment]
Just pushed a tag.

**Comment by greeny** [comment]
:+1: just used it in our code, thanks

**Comment by TomasVotruba** [comment]
:+1:
